### PR TITLE
[SW-1995] Enabling Compression for REST API Conversions

### PR DESCRIPTION
**Pull request by mn-mikke**
**Benchmark results on EMR** 

Compression `NONE`:
```
DataFrameToH2OFrameConversionBenchmark results for the dataset 'random01Int4':
time: 8964 milliseconds

DataFrameToH2OFrameConversionBenchmark results for the dataset 'random01Int40':
time: 49782 milliseconds

DataFrameToH2OFrameConversionBenchmark results for the dataset 'random01Int400':
time: 428360 milliseconds
```

Compression `SNAPPY`:
```
DataFrameToH2OFrameConversionBenchmark results for the dataset 'random01Int4':
time: 8915 milliseconds

DataFrameToH2OFrameConversionBenchmark results for the dataset 'random01Int40':
time: 48033 milliseconds

DataFrameToH2OFrameConversionBenchmark results for the dataset 'random01Int400':
time: 413359 milliseconds
```

Compression `DEFLATE`:
```
DataFrameToH2OFrameConversionBenchmark results for the dataset 'random01Int4':
time: 9702 milliseconds

DataFrameToH2OFrameConversionBenchmark results for the dataset 'random01Int40':
time: 49522 milliseconds

DataFrameToH2OFrameConversionBenchmark results for the dataset 'random01Int400':
time: 448095 milliseconds
```

Compression `GZIP`:
```
DataFrameToH2OFrameConversionBenchmark results for the dataset 'random01Int4':
time: 9386 milliseconds

DataFrameToH2OFrameConversionBenchmark results for the dataset 'random01Int40':
time: 50886 milliseconds

DataFrameToH2OFrameConversionBenchmark results for the dataset 'random01Int400':
time: 442188 milliseconds
```

### CODE REVIEW

File: py/src/ai/h2o/sparkling/ExternalBackendConf.py [hunk 1]
```diff
@@ -187,3 +190,7 @@ def setExternalHadoopExecutable(self, executable):
     def setExternalExtraJars(self, paths):
         self._jconf.setExternalExtraJars(paths)
         return self
+
```
Comment (jakubhava) [inline py/src/ai/h2o/sparkling/ExternalBackendConf.py]: I know the PR is still marked as WIP, but wanted to mention this before I forget. Would be good to expose it on H2OConf.R as well

Comment (mn-mikke) [reply py/src/ai/h2o/sparkling/ExternalBackendConf.py]: Will do, thank!

**Comment by mn-mikke** [comment]
@jakubhava It seems that heavy compressions like `DEFLATE` and `GZIP` are getting it worse. It would be great to run benchmarks also an a slower network infrastructure...

**Comment by jakubhava** [comment]
Interesting results though, I'm surprised the speed-up is not that obvious. But nevertheless, this is a good change!

**Comment by mn-mikke** [comment]
IMHO, the test cases are not a good input to compression algorithms. The values are made of 0/1 which are partially comressed by `AutoBuffer` anyway. I think we would get better results with a datasets containing, string/categorical columns. I will try to another test case in a different PR.

**Pull request merged**